src/main/scala/devices/debug/Debug.scala

```diff
@@ -569,7 +569,12 @@ class TLDebugModuleInner(device: Device, getNComponents: () => Int, beatBytes: I
     }
 
     val haltedSummary = Cat(haltedStatus.map(_.orR).reverse)
-    val HALTSUMRdData = (new HALTSUMFields()).fromBits(haltedSummary)
+    val HALTSUM1RdData = (new HALTSUM1Fields()).fromBits(haltedSummary)
+
+    val selectedHaltedStatus = Mux((selectedHartReg >> 5) >= (nComponents.U >> 5), 0.U, haltedStatus(selectedHartReg >> 5))
```

Comment:
Is Chisel smart enough to only generate a 5-bit comparator (instead of 10-bit) on this mux select (as opposed to if you'd coded the select as selectHartReg(10:5) >= nComponents(10:5))?